django/contrib/admindocs/utils.py

```diff
@@ -239,3 +242,57 @@ def remove_non_capturing_groups(pattern):
         final_pattern += pattern[prev_end:start]
         prev_end = end
     return final_pattern + pattern[prev_end:]
+
+
+# `ThreadCallbackStrs` object holds instances of the _callback_strs as c_strs
+# objects that are local to the thread.
+class ThreadCallbackStrs(threading.local):
```

Comment:
I forgot that we should be using `asgiref.Local` instead of `threading.local`. See `django/utils/translation/trans_real.py` for how it is implemented there for an idea of the pattern to follow.